### ISSUE TITLE
Crashes after interaction on Arch

### ISSUE THREAD

**Issue by chookity-pokk** [atlas-engineer/nyxt#2872]
**Describe the bug**
Was working just fine a few days ago, but I did an update yesterday and it's now not working. I launch Nyxt just fine, but once I try to look something up, I just get a white buffer, then after clicking any button it crashes with the below output in the terminal.

**Precise recipe to reproduce the issue**
Launch Nyxt, `C/M-l` type in something to search. Interact with the browser again, then it'll crash.

**Information**
- EndeavourOS
- Integrated graphics on an Intel i5-4300M (4) @ 3.300GHz 
- i3WM v4.22
- `sudo pacman -S nyxt`
- Information from `show-system-information`: 
```
Nyxt version: 2.2.4                                                                                                                                                    
Renderer version: GI-GTK                                                                                                                                               
Operating system kernel: Linux 6.2.7-arch1-1                                                                                                                           
Lisp implementation: SBCL 2.2.9 (Dynamic space size: 3221225472)                                                                                                       
Features: (:WEBKIT2 :WEBKIT2-2.38 :WEBKIT2-PASTE-PLAINTEXT :WEBKIT2-TRACKING
 :WEBKIT2-MUTE :WEBKIT2-EMOJI :WEBKIT2-MEDIA :WEBKIT2-SANDBOXING :GTK-3-22
 :GTK-3-20 :GTK-3-18 :GTK-3-16 :GTK-3-14 :GTK-3-12 :GTK-3-10 :GTK-3-8 :GTK-3-6
 :GTK-3-4 :GTK :GDK-3-22 :GDK-3-20 :GDK-3-18 :GDK-3-16 :GDK-3-14 :GDK-3-12
 :GDK-3-10 :GDK-3-8 :GDK-3-6 :GDK-3-4 :CAIRO-1-10 :CAIRO-1-12 :GDK-PIXBUF
 :GLIB-2-30 :GLIB-2-32 :GLIB-2-34 :GLIB-2-36 :GLIB-2-38 :GLIB-2-40 :GLIB-2-42
 :GLIB-2-44 :GLIB-2-46 :GLIB-2-48 :GLIB-2-50 :GLIB-2-52 :GLIB-2-54 :GLIB-2-56
 :GLIB-2-58 :GLIB :NYXT-2 :FSET-EXT-STRINGS :CUSTOM-HASH-TABLE-NATIVE :SWANK
 :PLUMP-UTF-32 :GLOBAL-VARS :DECLARE-TYPES :PARENSCRIPT
 :SBCL+SAFE-STANDARD-READTABLE :NAMED-READTABLES :LPARALLEL :21BIT-CHARS
 :CHUNGA :CLOSER-MOP :CL-PPCRE-UNICODE :FLEXI-STREAMS :CL-UNICODE :CL-PPCRE
 :CL-JSON-DOUBLE-FLOAT-IS-SUBSUMED :CL-JSON-SINGLE-FLOAT-IS-SUBSUMED
 :BORDEAUX-THREADS :LPARALLEL.WITH-CLTL2 :LPARALLEL.WITH-CAS
 :LPARALLEL.WITH-STEALING-SCHEDULER :SPLIT-SEQUENCE CHIPZ-SYSTEM:GRAY-STREAMS
 CFFI-FEATURES:FLAT-NAMESPACE CFFI-FEATURES:X86-64 CFFI-FEATURES:UNIX :CFFI
 CFFI-SYS::FLAT-NAMESPACE ALEXANDRIA::SEQUENCE-EMPTYP :FAST-IO-SV :FAST-IO
 :SBCL-USES-SB-ROTATE-BYTE :CL-JSON-CLOS :CL-JSON :THREAD-SUPPORT :ASDF3.3
 :ASDF3.2 :ASDF3.1 :ASDF3 :ASDF2 :ASDF :OS-UNIX :NON-BASE-CHARS-EXIST-P
 :ASDF-UNICODE :X86-64 :GENCGC :64-BIT :ANSI-CL :COMMON-LISP :ELF
 :IEEE-FLOATING-POINT :LINUX :LITTLE-ENDIAN :PACKAGE-LOCAL-NICKNAMES
 :SB-CORE-COMPRESSION :SB-LDB :SB-PACKAGE-LOCKS :SB-THREAD :SB-UNICODE :SBCL
 :UNIX)

ASDF version: 3.3.1
ASDF registries: (NYXT-SOURCE-REGISTRY ENVIRONMENT-SOURCE-REGISTRY)
Critical dependencies: (/build/nyxt/src/nyxt/_build/cl-cffi-gtk/gtk/cl-cffi-gtk.asd 
 /build/nyxt/src/nyxt/_build/cl-gobject-introspection/cl-gobject-introspection.asd
 /build/nyxt/src/nyxt/_build/cl-webkit/webkit2/cl-webkit2.asd)

```

**Output when started from a shell**
```
<INFO> [11:33:18] Listening to socket "/run/user/1000/nyxt/nyxt.socket".
Nyxt version 2.2.4

** (nyxt:13859): WARNING **: 11:33:19.135: WEBKIT_FORCE_SANDBOX no longer allows disabling the sandbox. Use WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS=1 instead.

** (nyxt:13859): ERROR **: 11:33:34.296: Unable to configure xdg-desktop-portal access in the WebKit sandbox: GApplication is required.
<WARN> [11:33:34] Warning: Error on GTK thread: The value
  :INVALID-CODE-OBJECT-AT-PC
is not of type
  (MOD 4611686018427387902)

** (process:13859): ERROR (recursed) **: Unable to configure xdg-desktop-portal access in the WebKit sandbox: GApplication is required.
fatal error encountered in SBCL pid 13859 tid 13871:
SIGABRT received.

   0: fp=0x7f209ea476c0 pc=0x7f20a8dcc8ec Foreign function (null)
```

**Comment by fold-squirrel** [comment]
I have the same problem as well, I'm using arch linux. reading the error message it seems you need to set the environment variable `WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS` to `1` for webkit to disable the sandbox.
so running
> `WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS=1 nyxt`
works just fine

**Comment by jmercouris** [comment]
Something tells me doing so is dangerous... :-D

I would *highly* suggest if you are running an older version of Nyxt, to run it in a sandbox of some sort!

**Comment by chookity-pokk** [comment]
Are pre-release releases pushed to repos? If not then I believe I am up to date. Though my desktop also has version 2.2.4 and that seems to be working just fine! Same OS and setup.

**Comment by jmercouris** [comment]
Pre-releases are technically tagged releases.

**Comment by marcelofern** [comment]
Same error here, I'm also in Arch. The flag worked, though it doesn't look like a choice I want to commit to!

**Comment by aadcg** [comment]
Could you report which exact version of webkitgtk are you running? Is it 2.38.5? Thanks.

**Comment by aartaka** [comment]
It's not dependent on WebKit, it's our `libraries/web-extensions` being both broken and sandbox-violating.

See #2177 for more context and suggestion for 3.0.

@marcelofern, @fold-squirrel, there are three options for you that I can see:
- Least resistance, but dangerous: use `WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS`. 
- More involved, but safer: find `libnyxt.so` somewhere in installed Nyxt directories and deleting it.
- Slightly more involved: install pre-releases for version 3 (especially the latest ones, they are pretty close to being stable), because the library is doing nothing and is absolutely safe to delete.

**Comment by chookity-pokk** [comment]
Sorry for the late reply, I can confirm that building the latest release works just fine on Arch! Also, the default homepage looks really nice in 3.0! Great work to all the devs!

**Comment by aartaka** [comment]
Closing then, as it's fixed on master and there are ways to fix it on 2.2.4 too.
